Source/ProgressionSystemRuntime/Private/Components/PSHUDComponent.cpp

```diff
@@ -195,13 +191,14 @@ void UPSHUDComponent::UpdateProgressionWidgetForPlayer()
 //Is called when local player character is ready to guarantee that they player controller is initialized for the Widget SubSystem
 void UPSHUDComponent::OnLocalCharacterReady_Implementation(APlayerCharacter* Character, int32 CharacterID)
 {
-	// Create widgets now as fast as possible, later we will register them in Widgets Subsystem
-	UWidgetsSubsystem& WidgetsSubsystem = UWidgetsSubsystem::Get(Character);
```

Comment:
The WidgetsSubsystem requires OnLocalCharacterReady usage to ensure widgets are initialized only when a local character is ready, for safety it shouldn't be used anywhere else

Why:
- Using your OnInitialized you create dependency order between the subsystem and widgets.
- Using OnLocalCharacterReady ensures widgets are created as soon as local character is ready, aligning with clear, dependency-aware initialization.

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/41/commits/67874141fbb83e944ed2b656e4a835052e280812